clippy_lints/src/static_items_large_align.rs

```diff
@@ -0,0 +1,186 @@
+use clippy_utils::diagnostics::span_lint_and_then;
+use rustc_data_structures::fx::FxHashSet;
+use rustc_hir::{Item, ItemKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty::{self, AdtDef, Ty, TyCtxt, TypeAndMut};
+use rustc_session::{declare_tool_lint, impl_lint_pass};
+use rustc_span::Span;
+use rustc_typeck::hir_ty_to_ty;
+
+pub struct StaticItemsLargeAlign {
+    pub page_size: u64,
+}
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Check for usage of static items which have type alignment larger than page size.
+    ///
+    /// ### Why is this bad?
+    /// Due to some known unsound issues, the type alignment may not be fulfilled.
+    /// For more information, see:
+    /// <https://github.com/rust-lang/rust/issues/70022> and
+    /// <https://github.com/rust-lang/rust/issues/70143>.
+    ///
+    /// ### Example
+    /// ```rust
+    /// #[repr(align(0x100000))]
+    /// struct Aligned(u8);
+    ///
+    /// static X: Aligned = Aligned(0);         // Bad
+    ///
+    /// fn main() {
+    ///     let x = Aligned(0);                 // Good
+    ///     println!("{:#x}", &x as *const _ as usize);
+    ///     println!("{:#x}", &X as *const _ as usize);
+    ///     let b = Box::new(Aligned(0));       // Good
+    ///     println!("{:#x}", Box::into_raw(b) as usize);
+    /// }
+    /// ```
+    #[clippy::version = "1.61.0"]
```

Comment:
```suggestion
    #[clippy::version = "1.66.0"]
```

This will most likely be merged in the 1.66 cycle.